languages/elixir/exercises/concept/binary-matching/.meta/example.ex

```diff
@@ -0,0 +1,24 @@
+defmodule FileSniffer do
+  def type_from_extension(<< "bmp" >>), do: {:ok, "image/bmp"}
+  def type_from_extension(<< "png" >>), do: {:ok, "image/png"}
+  def type_from_extension(<< "jpg" >>), do: {:ok, "image/jpg"}
+  def type_from_extension(<< "gif" >>), do: {:ok, "image/gif"}
+  def type_from_extension(<< "exe" >>), do: {:ok, "application/octet-stream"}
+
+  def type_from_header(<< 0x42, 0x4D, _::binary >>), do: {:ok, "image/bmp"}
+  def type_from_header(<< 0x89, 0x50, 0x4E, 0x47, 0x0D, 0x0A, 0x1A, 0x0A, _::binary >>), do: {:ok, "image/png"}
+  def type_from_header(<< 0xFF, 0xD8, 0xFF, _::binary >>), do: {:ok, "image/jpg"}
+  def type_from_header(<< 0x47, 0x49, 0x46, _::binary >>), do: {:ok, "image/gif"}
+  def type_from_header(<< 0x7F, 0x45, 0x4C, 0x46, _::binary >>), do: {:ok, "application/octet-stream"}
```

Comment:
It's hard to judge the example solution without the instructions, but if I were writing this code for real-life purposes, I would definitely try to write the printable characters using `?`, e.g. `<< 0x7F, ?E, ?L, ?F, _::binary >`. All of those hexadecimal numbers tell me nothing when I look at them. On the other hand, if the instructions just tell you the numbers in hexadecimal and don't mention that some of those are letters, I guess it's fine.

Reply:
I think I will spell out the file headers in the exercise instructions. I wanted a chance to highlight integer literals of different bases.

I also went with these because when I google `elf file format` (and similarly for bmp/gif/jpg/png) and pull up the [wiki](https://en.wikipedia.org/wiki/Executable_and_Linkable_Format) the hex dump is either fully hex or the ascii (with unprintable character) representation.